src/platforms/esp32/main/spi_driver.c

```diff
@@ -249,8 +251,8 @@ static void spidriver_consume_mailbox(Context *ctx)
             ret = ERROR_ATOM;
     }
 
-    mailbox_destroy_message(message);
+    term ret_msg = port_create_tuple2(ctx, term_get_tuple_element(msg, 1), ret);
```

Comment:
I suggest assigning term_get_tuple_element to a temporary variable, in order to give it a meaningful name.

Reply:
At the moment I feel like that we might deprecate port_create_tuple2 for 2 reasons:
- it hides memory allocation and a potential garbage collection
- memory allocation may fail and it doesn't allow to handle it